klippy/extras/bed_mesh.py

```diff
@@ -653,8 +751,17 @@ def cmd_BED_MESH_CALIBRATE(self, gcmd):
         self._profile_name = gcmd.get('PROFILE', "default")
         if not self._profile_name.strip():
             raise gcmd.error("Value for parameter 'PROFILE' must be specified")
+        exclude_object = None
+        adaptive_mesh = bool(gcmd.get_int('ADAPTIVE', 0))
+        margin = gcmd.get_float('ADAPTIVE_MARGIN', self.bedmesh.margin)
+        if adaptive_mesh:
+            try:
+                exclude_object = self.printer.lookup_object("exclude_object")
```

Comment:
> I would recommend calling after the ALGORITHM option is checked, as we don't want any other parameters to override the adaptive mesh.

I added the adaptive logic before the rest of the parameter processing in order to allow parameters specified on the `BED_MESH_CALIBRATE` command line to override the adaptive settings. If the adaptive logic was called after, would the adaptive params override `BED_MESH_CALIBRATE MESH_MIN=x,y`?

Reply:
Yes, if the adaptive logic is added after it will override other params.  I think this makes sense, as that is the point of activating the adaptive mesh.  Mixing parameters with the adaptive logic would seem to produce unexpected results.

Reply:
Done in latest revision.